tests/test_api.py

```diff
@@ -1485,3 +1545,217 @@ def test_operations(self):
+            api.move_project(self.ctx,
+                             11111111,
+                             to_project_id=self.parent_project.id)
+
+        # Check if there are no transactions created when there is an error
+        transactions = Transaction.objects.filter(created_at__gte=timestamp)
+        self.assertEqual(len(transactions), 0)
+
+    def test_to_project_not_exists(self):
+        """Check if it fails when the destination parent project does not exist"""
+
+        timestamp = datetime_utcnow()
+
+        with self.assertRaisesRegex(NotFoundError, PROJECT_NOT_FOUND_ERROR.format(proj_id='11111111')):
+            api.move_project(self.ctx,
+                             self.project.id,
+                             to_project_id=11111111)
+
+        # Check if there are no transactions created when there is an error
+        transactions = Transaction.objects.filter(created_at__gte=timestamp)
+        self.assertEqual(len(transactions), 0)
+
+    def test_to_project_invalid(self):
+        """Check if it fails when the destination project is the same as the source"""
+
+        timestamp = datetime_utcnow()
+
+        with self.assertRaisesRegex(InvalidParentError, PROJECT_INVALID_PARENT_EQUAL_ERROR):
+            api.move_project(self.ctx,
+                             self.project.id,
+                             to_project_id=self.project.id)
+
+        # Check if there are no transactions created when there is an error
+        transactions = Transaction.objects.filter(created_at__gte=timestamp)
+        self.assertEqual(len(transactions), 0)
+
+    def test_parent_different_root_project(self):
+        """Check if it fails when trying set as parent a project from a different root project"""
+
+        root1 = api.add_project(self.ctx,
+                                ecosystem_id=self.origin_eco.id,
+                                name='root-1')
+        child1 = api.add_project(self.ctx,
+                                 ecosystem_id=self.origin_eco.id,
+                                 name='example-child',
+                                 parent_id=root1.id)
+
+        timestamp = datetime_utcnow()
+
+        with self.assertRaisesRegex(InvalidParentError, PROJECT_INVALID_PARENT_DIFFERENT_ROOT):
+            api.move_project(self.ctx,
+                             child1.id,
+                             self.parent_project.id)
+
+        # Check if there are no transactions created when there is an error
+        transactions = Transaction.objects.filter(created_at__gte=timestamp)
+        self.assertEqual(len(transactions), 0)
+
+    def test_set_child_as_parent(self):
+        """Check if it fails when trying set as parent a child project"""
+
+        timestamp = datetime_utcnow()
+
+        with self.assertRaisesRegex(InvalidParentError, PROJECT_INVALID_PARENT_EQUAL_ERROR):
+            api.move_project(self.ctx,
+                             self.project.id,
+                             self.project.id)
```

Comment:
Is this the best example we can have? Probably a better test is to use a grandparent-parent-child relationship and try to move the parent under its child.